cmd/tether/attach.go

```diff
@@ -518,16 +597,58 @@ func (t *attachServerSSH) globalMux(reqchan <-chan *ssh.Request) {
 func (t *attachServerSSH) channelMux(in <-chan *ssh.Request, session *tether.SessionConfig, cleanup func()) {
 	defer trace.End(trace.Begin("attach server channel request handler"))
 
-	// for the actions after we process the request
-	var pendingFn func()
-
 	// cleanup function passed by the caller
 	defer cleanup()
 
+	// to make sure we close the channel once
+	var once sync.Once
+
+	// for the actions after we process the request
+	var pendingFn func()
 	for req := range in {
 		ok := true
+		abort := false
+
+		log.Infof("received channel mux type %v", req.Type)
 
 		switch req.Type {
+		case msgs.PingReq:
+			log.Infof("Received PingReq for %s", session.ID)
+
+			if string(req.Payload) != msgs.PingMsg {
+				log.Infof("Received corrupted PingReq for %s", session.ID)
+				ok = false
+			}
+		case msgs.UnblockReq:
+			log.Infof("Received UnblockReq for %s", session.ID)
+
+			if string(req.Payload) != msgs.UnblockMsg {
```

Comment:
Nope I guess it is just defensive programming, could drop those if you think they are unnecessary